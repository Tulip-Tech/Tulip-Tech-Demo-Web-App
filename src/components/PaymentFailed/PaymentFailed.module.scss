@use 'src/styles/variables';
@use 'src/styles/theme';

.title {
  font-weight: var(--body-font-weight-bold);
  font-size: 24px;
}

.message {
  font-size: 16px;
}
