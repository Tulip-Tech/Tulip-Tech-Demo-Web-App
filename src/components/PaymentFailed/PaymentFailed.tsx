import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './PaymentFailed.module.scss';

import Button from '#components/Button/Button';

type Props = {
  type: 'error' | 'cancelled';
  message?: string;
  onCloseButtonClick: () => void;
};

const PaymentFailed: React.FC<Props> = ({ type, message, onCloseButtonClick }: Props) => {
  const { t } = useTranslation('account');
  return (
    <div>
      <h2 className={styles.title}>{type === 'cancelled' ? t('checkout.payment_cancelled') : t('checkout.payment_error')}</h2>
      <p className={styles.message}>{type === 'cancelled' ? t('checkout.payment_cancelled_message') : message}</p>
      <div>
        <Button label={t('checkout.close')} onClick={onCloseButtonClick} color="primary" variant="contained" size="large" fullWidth />
      </div>
    </div>
  );
};

export default PaymentFailed;
