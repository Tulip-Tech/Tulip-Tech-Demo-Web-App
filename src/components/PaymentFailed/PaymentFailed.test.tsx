import React from 'react';
import { render } from '@testing-library/react';

import PaymentFailed from './PaymentFailed';

describe('<PaymentFailed>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<PaymentFailed type="error" onCloseButtonClick={vi.fn()} message="Error message" />);

    expect(container).toMatchSnapshot();
  });
});
