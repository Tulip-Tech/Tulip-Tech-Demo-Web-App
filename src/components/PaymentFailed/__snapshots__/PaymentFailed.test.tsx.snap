// Vitest Snapshot v1

exports[`<PaymentFailed> > renders and matches snapshot 1`] = `
<div>
  <div>
    <h2
      class="_title_957731"
    >
      checkout.payment_error
    </h2>
    <p
      class="_message_957731"
    >
      Error message
    </p>
    <div>
      <button
        class="_button_f8f296 _primary_f8f296 _fullWidth_f8f296 _large_f8f296"
      >
        <span>
          checkout.close
        </span>
      </button>
    </div>
  </div>
</div>
`;
